java/core/lance-jni/src/blocking_dataset.rs

```diff
@@ -42,6 +43,11 @@ impl BlockingDataset {
         Ok(Self { inner })
     }
 
+    pub fn commit(uri: &str, operation: Operation, read_version: Option<u64>) -> Result<Self> {
+        let inner = RT.block_on(Dataset::commit(uri, operation, read_version, None, None))?;
```

Comment:
TODO commit without read_version will throw Unknown error which is hard for downstream to debug